api/envoy/config/listener/v3/listener.proto

```diff
@@ -138,6 +139,9 @@ message Listener {
   // :ref:`FAQ entry <faq_how_to_setup_sni>`.
   repeated FilterChain filter_chains = 3;
 
+  // FCDS: Filter Chain Discovery Service config block. filter_chains and fcds are mutually exclusive
+  Fcds fcds = 34;
```

Comment:
I think this needs to be promoted to oneof with `filter_chains`, and clearly comment that they are mutual exclusive.

Also, is this going to return a single filter-chain? Maybe it should return a glob collection of filter-chains.

Reply:
I tried oneof when I started implementing this. Had some issues that I can't recall. Will give it another try and update this PR as needed.

Reply:
> I tried oneof when I started implementing this. Had some issues that I can't recall. Will give it another try and update this PR as needed.
> 
Changing to oneof breaks backwards compatibility. This should be promoted in future major version, and clearly stated in the comments.


> FCDS block would encompass a collection of filter chains., which are under a listener Plz refer to the yaml samples in the PR description.

Looking [here](https://github.com/envoyproxy/envoy/pull/23096/files#diff-cc79aca7dddd943b7bb9979041523e3a1ffd52c36a6f51b5265493b828b48572R27) it implies that the returned type is a single FilterChain object for each request, not multiple ones. I think this should be a [Glob collection](https://github.com/cncf/xds/blob/main/proposals/TP1-xds-transport-next.md#glob) and return a list of such elements.

Reply:
Hi @kyessenov, @adisuissa 
I gave it a thought. I want to represent a single resource as a filter chain, not a list of filter chains. I think the name that I added in the proto, created confusion, which I could remove as it's not used in any logic.

- For file inotify, much like LDS, I wanted a single listener to point to one FCDS file which has a list of filter chains resources
- Also, for GRPC, I want a listener to be able to subscribe to all(*) or a specific list of filter chains. The Control plane would respond back with a list of resources, where each resource is a filter chain.

In that case, much like LDS proto, I believe the resource type not being a list is fine. 

However, the challenge here is, how could a listener subscribe to a specific list of filter chain updates. I dont see a way set the resource names in the config_source proto. Is there any example of xDS that I can refer to?

Reply:
> Changing to oneof breaks backwards compatibility. This should be promoted in future major version, and clearly stated in the comments.
> 
> >

@adisuissa  - Can you please show an example how to promote a proto change to a "Future major version"?